code/game/gamemodes/miniantags/abduction/abduction.dm

```diff
@@ -20,191 +15,59 @@
 	to_chat(world, "<b>Crew</b> - don't get abducted and stop the abductors.")
 
 /datum/game_mode/abduction/pre_setup()
-	possible_abductors = get_players_for_role(ROLE_ABDUCTOR)
-
+	var/possible_abductors = get_players_for_role(ROLE_ABDUCTOR)
 	if(!length(possible_abductors))
-		return 0
-
-	abductor_teams = max(1, min(max_teams,round(num_players()/15)))
-	var/possible_teams = max(1,round(length(possible_abductors) / 2))
-	abductor_teams = min(abductor_teams,possible_teams)
-
-	abductors.len = 2*abductor_teams
-	scientists.len = abductor_teams
-	agents.len = abductor_teams
-	team_objectives.len = abductor_teams
-	team_names.len = abductor_teams
-
-	for(var/i=1,i<=abductor_teams,i++)
-		if(!make_abductor_team(i))
-			return 0
-	..()
-	return 1
-
-/datum/game_mode/abduction/proc/make_abductor_team(team_number,preset_agent=null,preset_scientist=null)
-	//Team Name
-	team_names[team_number] = "Mothership [pick(GLOB.greek_letters)]" //TODO Ensure unique and actual alieny names
-	//Team Objective
-	var/datum/objective/experiment/team_objective = new
-	team_objective.abductor_team_number = team_number
-	team_objectives[team_number] = team_objective
-	//Team Members
-
-	if(!preset_agent || !preset_scientist)
-		if(length(possible_abductors) <=2)
-			return 0
-
-	var/datum/mind/scientist
-	var/datum/mind/agent
+		return FALSE
 
-	if(!preset_scientist)
-		scientist = pick(possible_abductors)
-		possible_abductors -= scientist
-	else
-		scientist = preset_scientist
+	abductor_teams = max(1, min(min(max_teams, round(num_players() / 15)), round(length(possible_abductors) / 2)))
```

Comment:
Use `clamp(value, min, max)` to clean this up a bit. Looks like we'll still need a min after that, but it'll be more readable even so.

Reply:
Changes applied, PTAL

Reply:
Better, thanks.